go_modules/lib/dependabot/go_modules/update_checker.rb

```diff
@@ -83,23 +81,9 @@ def version_from_tag(tag)
         tag&.fetch(:tag)
       end
 
-      def git_dependency?
-        git_commit_checker.git_dependency?
```

Comment:
The base class implements `existing_version_is_sha?`  by delegating to `git_dependency?`. Would it make more sense to remove `existing_version_is_sha?` and leave `git_dependency?` implemented as `false`?

I guess not technically correct since from your explanation, go _does_ have git dependencies, they just map to "standard versions". But just concerned that we could copy some code from another ecosystem in the future that has branching according to `git_dependency?`, and we unintentionally reintroduce this kind of bug.

ANyways, no big deal, this is great!

Reply:
Yeah it's confusing, it does have Git dependencies... but they aren't SHAs. Since they're comparable, the reasoning why we treated Git dependencies as pins in other ecosystems doesn't apply here. 

The base class doesn't use `git_dependency?` though? 

https://github.com/dependabot/dependabot-core/blob/7196bb00a2e52d0d2ed167d1964ddab7e88dc1fe/common/lib/dependabot/update_checkers/base.rb#L207-L211

I did consider removing `git_dependency?` completely since the base class implementation will always be false, but I liked putting a comment above the method. 🤷

Reply:
Disregard my previous comment 😳. I greped for the base class implementation, and I confused the previous implementation in the go ecosystem, with the base class implementation 😆